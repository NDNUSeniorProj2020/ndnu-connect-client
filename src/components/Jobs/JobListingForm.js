import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Input, Form, Radio, Button } from 'antd';
import MarkdownIt from 'markdown-it';
import MarkdownEditor from 'react-markdown-editor-lite';
import 'react-markdown-editor-lite/lib/index.css';
        const styles = { width: '50%' };
        const mdParser = new MarkdownIt();
    export default function JobListingForm({ onSubmit }) {
        const [title, setTitle] = useState('');
        const [company, setCompany] = useState('');
        const [location, setLocation] = useState('');
        const [description, setDescription] = useState('');
        const [type, setType] = useState('');
        const onSubmit = (e) => {
            e.preventDefault();
            const job = { title, company, location, description, type };
            onSubmit(job);
          };
    return (
        <Form id="job-listing-form" onSubmit={onSubmit}>
        <Form.Item label="Title" name="title">
        <Input
        id="title-input"
        style={styles}
        value={title}
        onChange={e => setTitle(e.target.value)}
        placeholder="Enter job title."
        />
        </Form.Item>
        <Form.Item label="Company Name" name="company">
        <Input
        id="company-input"
        style={styles}
        value={company}
        onChange={e => setCompany(e.target.value)}
        placeholder="Enter company name."
        />
        </Form.Item>
        <Form.Item label="Loaction" name="location">
        <Input
        id="location-input"
        style={styles}
        value={location}
        onChange={e => setLocation(e.target.value)}
        placeholder="Enter location."
        />
        </Form.Item>
        <Form.Item label="Description" name="description">
        <MarkdownEditor
        id="description-input"
        style={{ ...styles, height: '400px' }}
        value={description}
        renderHTML={text => mdParser.render(text)}
        onChange={parser => setDescription(parser.text)}
        />
        </Form.Item>
        <Form.Item label="Type" name="type">
        <Radio.Group id="type-input" onChange={e => setType(e.target.value)} value={type}>
        <Radio value="FULL">Full-Time</Radio>
        <Radio value="PART">Part-Time</Radio>
        <Radio value="INTR">Internship</Radio>
        </Radio.Group>
        </Form.Item>
        <Form.Item>
        <Button className="submit-button" type="primary" onClick={onSubmit}>Submit</Button>
        </Form.Item>
        </Form>
        );
        }