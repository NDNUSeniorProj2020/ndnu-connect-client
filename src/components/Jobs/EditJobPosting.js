import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { message } from 'antd';

import { fetchJob, updateJob } from '../../actions/jobs/jobsActions';
import JobListingForm from './JobListingForm';

export function ConnectedEditJobListing({ job, success, updated, fetchJob, updateJob }) {
  useEffect(() => {
    const id = window.location.pathname.replace('/jobs/edit/', '');
    const token = localStorage.getItem('token');

    fetchJob(token, id);
  }, [fetchJob]);

  const editJob = async (editedJob) => {
    const updatedJob = Object.assign({}, job, editedJob)

    try {
      await updateJob(localStorage.getItem('token'), updatedJob);
    } catch (err) {
      console.error(err);
      message.error('Cannot update job listing. Sorry for the inconvenience. Please try again.', 60);
    }
  };

  if (updated)
    return <Redirect to="/jobs" />;

  if (success && Object.keys(job).length > 0)
    return (
      <div>
        <header>
          <h1>Edit Job Listing</h1>
        </header>
        <JobListingForm job={job} submitJob={editJob} />
      </div>
    );

  return <p>Loading page...</p>;
}

ConnectedEditJobListing.propTypes = {
  job: PropTypes.object,
  success: PropTypes.bool,
  updated: PropTypes.bool,
  updateJob: PropTypes.func,
  fetchJob: PropTypes.func
};
ConnectedEditJobListing.defaultProps = {
  job: {},
  success: false,
  updated: false,
  updateJob: f => f,
  fetchJob: f => f
};

const mapStateToProps = ({ jobsReducer }) => ({
  job: jobsReducer.job,
  success: jobsReducer.success,
  updated: jobsReducer.updated
});
const EditJobListing = connect(mapStateToProps, { fetchJob, updateJob })(ConnectedEditJobListing);

export default EditJobListing;
