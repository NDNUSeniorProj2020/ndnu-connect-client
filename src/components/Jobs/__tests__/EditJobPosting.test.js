import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter, Redirect } from 'react-router-dom';

import { ConnectedEditJobListing } from '../EditJobPosting';
import JobListingForm from '../JobListingForm';

const job = {
  'id': 1,
  'title': 'test1',
  'description': 'description test',
  'qualifications': 'bs degree',
  'pay': '1234',
  'link': 'google.com',
  'date': '2020-03-13T11:55:20.710240-07:00',
  'type': 'FULL',
  'user': 1
};

describe('tests for EditJobListing components', () => {
  describe('tests for ConnectedEditJobListing', () => {
    let wrapper;
    let props;

    beforeEach(() => {
      props = {
        job,
        success: true,
        updateJob: jest.fn()
      };
      wrapper = mount(
        <BrowserRouter>
          <ConnectedEditJobListing {...props} />
        </BrowserRouter>
      );
    });

    describe('snapshot tests', () => {
      it('renders without crashing', () => expect(wrapper).toMatchSnapshot());
    });

    describe('unit tests', () => {
      it('has default props', () => {
        const { success, updateJob, fetchJob, updated } = ConnectedEditJobListing.defaultProps;
        expect(success).toBe(false);
        expect(updated).toBe(false);
        expect(fetchJob('f')).toEqual('f');
        expect(updateJob('f')).toEqual('f');
      });

      it('renders Loading page if the page has not loaded', () => {
        wrapper = mount(
          <BrowserRouter>
            <ConnectedEditJobListing />
          </BrowserRouter>
        );
        expect(wrapper.find('p').text()).toEqual('Loading page...');
      });
    });

    describe('integration tests', () => {
      it('saves job on submit from JobListingForm', () => {
        wrapper.find(JobListingForm).props().submitJob(job);
      });

      it('redirects if updating a job posting was successful', () => {
        wrapper = mount(
          <BrowserRouter>
            <ConnectedEditJobListing {...props} updated={true} />
          </BrowserRouter>
        );
        expect(wrapper.find(Redirect)).toBeDefined();
      });
    });
  });
});
