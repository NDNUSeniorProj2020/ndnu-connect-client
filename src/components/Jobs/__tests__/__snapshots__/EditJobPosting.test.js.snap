// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for EditJobListing components tests for ConnectedEditJobListing snapshot tests renders without crashing 1`] = `ReactWrapper {}`;
